apps/api/src/domain/games/memory-match/services/memory-match.service.ts

```diff
@@ -29,23 +33,38 @@ export class MemoryMatchService {
         }
 
         // our convention is to return not published when a user does not have access to a particular resource
-        if (!searchResult.isPublished) {
+        if (isNullOrUndefined(userWithGroups) && !searchResult.isPublished) {
+            return NotFound;
+        }
+
+        if (
+            !isNullOrUndefined(userWithGroups) &&
+            !userWithGroups.isAdmin() &&
+            !searchResult.isPublished
+        ) {
             return NotFound;
         }
 
         // at this point, we have a published round
         return this.buildView(searchResult);
     }
 
-    async fetchMany(): Promise<{ entities: IMemoryMatchRound[] }> {
+    async fetchMany(
+        userWithGroups?: CoscradUserWithGroups
+    ): Promise<{ entities: IMemoryMatchRound[] }> {
         // TODO filter out unpublished round at the level of the database
         const searchResult = await this.memoryMatchRepository.fetchMany();
 
+        const availableEntities = searchResult.filter((entity) => {
```

Comment:
Use the original `flatmap` pattern or else a single `reduce`. If we have a large number of rounds, this becomes a performance issue at some point. I'm trying to get away from long `map\reduce\filter` chains if I know the size of the array could be big. JS has to create and destroy a large number of cloned objects for each pass unnecessarily. I wish we had a language that could "roll up" these chains for us, because I do prefer the readability\ composability aspect.